### PR TITLE
feat(templates/lnodetype-wizards): add helper wizard for missing lnClass in templates

### LINKED ISSUE #223: LNodeType creation helper wizards
As a user of OpenSCD I want to be guided through the process of creating a new `LNodeType` element. I want the software to tell me which data objects `DO` are mandatory for the logical node and which common data classes the data object is based on. 

Two wizards are required here:
1. Choose the logical node class. Array automatically read from `IEC_61850-7-4_2007B3`. Bases on logical node class, display possible data objects in a selectable list. Mandatory objects shall be preselected.
2. Second wizard is the `dOTypeWizard` where the user shall have the possibility to select the CDC. CDCs shall be preselected based on the information in `IEC_61850-7-4_2007B3`.

**Features**
- Array of possible logical node classes are based on the definition in `IEC_61850-7-4_2007B3`
- Array of possible type in dOTypeWizard are based on the CDC type definition in `IEC_61850-7-4_2007B3` and defined DOTypes in the DataTypeTemplates section
- Progress shall be indicated in terms of how many wizards follow
- When canceling the process, no actions shall be performed. This is especially important when canceling in dOTypeWizard

### PR DESCRIPTION
**Pull request by JakobVogelsang**
Closes #223

### CODE REVIEW

**Comment by JakobVogelsang** [comment]
@ca-d  I slightly changed this task and introduced an `autoimport` option. With this option, we seperate the import from `templates` from the guided creation based on `*.nsd` files.

File: src/editors/templates/lnodetype-wizard.ts [hunk 1]
```diff
@@ -231,85 +232,233 @@ function dOWizard(options: WizardOptions): Wizard | undefined {
   ];
 }
 
+function getAdjacentClass(nsd74: XMLDocument, base: string): Element[] {
```
Comment (ca-d) [inline src/editors/templates/lnodetype-wizard.ts:235]: Maybe rename to `getDescendantClasses`?

File: src/editors/templates/lnodetype-wizard.ts [hunk 2]
```diff
@@ -231,85 +232,233 @@ function dOWizard(options: WizardOptions): Wizard | undefined {
...
+  return adjacents.concat(
+    Array.from(
+      nsd74.querySelectorAll(
+        `LNClass[name="${base}"], AbstractLNClass[name="${base}"]`
+      )
+    )
+  );
+}
+
+function getAllDataObject(nsd74: XMLDocument, base: string): Element[] {
```
Comment (ca-d) [inline src/editors/templates/lnodetype-wizard.ts:252]: `getAllDataObjects`

File: src/editors/templates/lnodetype-wizard.ts [hunk 3]
```diff
@@ -231,85 +232,233 @@ function dOWizard(options: WizardOptions): Wizard | undefined {
...
+      parent,
+      element: newLNodeType,
+      reference: getReference(parent, 'LNodeType'),
+    },
+  });
+
+  return unifyCreateActionArray(actions);
+}
+
+function proceedLNodeTypeCreate(
```
Comment (ca-d) [inline src/editors/templates/lnodetype-wizard.ts:355]: Maybe `processLNodeTypeCreate` or `startLNodeTypeCreate`?

File: src/editors/templates/lnodetype-wizard.ts [hunk 4]
```diff
@@ -231,85 +232,233 @@ function dOWizard(options: WizardOptions): Wizard | undefined {
...
+      newWizardEvent(createLNodeTypeHelperWizard(parent, newLNodeType, allDo))
+    );
 
-    return unifyCreateActionArray(actions);
+    wizard.dispatchEvent(newWizardEvent());
+    return [];
   };
 }
 
+function manageAutoimport(e: Event, templates: XMLDocument): void {
```
Comment (ca-d) [inline src/editors/templates/lnodetype-wizard.ts:403]: `togglePredefinedUI` or `togglePredefinedSwitch`

**Review by ca-d — APPROVED** [review]
LGTM

**Comment by ca-d** [comment]
We could also:

* move `dialogInitialFocus` to LN Class `mwc-select`
* remove numbers from LN CLass select
* sort lowercased string matching `DOType`s upwards in select menu
* sort DOType selects themselves by required
* move the predefined template toggle button to the bottom and disable it instead of hiding it

**Pull request merged**